### PR TITLE
feat(kv): Introduce 'DirEntry' realted codes

### PR DESCRIPTION
**Pull request by xiaguan**
For the subsequent folder restructuring, in this PR, we have done some preliminary work for refactoring. We have added new key/value `direntry` and also restructured the structure of the `kv_engine` module.

### CODE REVIEW

File: src/async_fuse/memfs/direntry.rs [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    #[must_use]
+    pub fn get_name(&self) -> &str {
+        &self.name
+    }
+
+    /// Returns the type of the file (directory, file, or symlink).
+    #[must_use]
+    pub fn get_file_type(&self) -> FileType {
+        self.file_type.clone()
+    }
```
Comment (Reto911) [inline src/async_fuse/memfs/direntry.rs:97]: Getters in Rust are not with the prefix `get_` conventionally, just name them as `inum()`, `name()` and `file_type()`.

**Comment by Reto911** [comment]
BTW, we serialize values into JSON now, but you serialized `DirEntry` with `bincode` in your test cases, are you going to use `bincode` for all kinds of values in the future?

File: src/async_fuse/memfs/kv_engine/mod.rs [hunk 2]
```diff
@@ -1,239 +1,28 @@
...
 
-use super::s3_node::S3Node;
-use super::s3_wrapper::S3BackEnd;
-use super::{INum, S3MetaData};
-use crate::async_fuse::memfs::dist::id_alloc::IdType;
 use crate::common::async_fuse_error::KVEngineError;
 use crate::common::error::{DatenLordError, DatenLordResult};
 
 /// The `KVEngineType` is used to provide support for metadata.
 /// We use this alias to avoid tem
```
[the remark this answers is not in the document]
Comment (xiaguan) [reply src/async_fuse/memfs/kv_engine/mod.rs:11]: template

File: src/async_fuse/memfs/kv_engine/key_type.rs [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+            KeyType::VolumeInfo(_) => "VolumeInfo",
+            KeyType::FileNodeList(_) => "FileNodeList",
+        }
+    }
+}
+
+impl LockKeyType {
+    /// Get the key in vec bytes.
+    #[must_use]
+    pub fn get_key(&self) -> String {
```
[the remark this answers is not in the document]
Comment (xiaguan) [reply src/async_fuse/memfs/kv_engine/key_type.rs:105]: I will rename to `to_string_key`.

File: src/async_fuse/memfs/kv_engine/key_type.rs [hunk 4]
```diff
@@ -0,0 +1,212 @@
...
+            KeyType::INum2Node(ref inum) => write!(f, "{inum}"),
+            KeyType::DirEntryKey((ref paretn_id, ref child_name)) => {
+                write!(f, "{paretn_id}_{child_name}")
+            }
+            #[cfg(test)]
+            KeyType::String(ref s) => write!(f, "{s}"),
+            KeyType::IdAllocatorValue(ref id_type) => write!(f, "{id_type}"),
+            KeyType::NodeIpPort(ref s) => write!(f, "{s}"),
+            KeyType::VolumeInfo(ref s) => write!(f, "{s}"),
+            KeyType::FileNodeList(ref s) => write!(f, "{s}"),
```
[the remark this answers is not in the document]
Comment (xiaguan) [reply src/async_fuse/memfs/kv_engine/key_type.rs:58]: It's not serialization. It's just kind of format.
I'll split it to another function.

File: src/async_fuse/memfs/direntry.rs (hunk 1 above)
Comment (xiaguan) [reply src/async_fuse/memfs/direntry.rs:97]: Ok, I'll fix it.

**Comment by xiaguan** [comment]
> BTW, we serialize values into JSON now, but you serialized `DirEntry` with `bincode` in your test cases, are you going to use `bincode` for all kinds of values in the future?

Yes, I'll use `bincode` for all kinds. But I will leave json format with daed_code. Use json is much easier to debug.

File: src/async_fuse/memfs/kv_engine/mod.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (Reto911) [reply src/async_fuse/memfs/kv_engine/mod.rs:11]: Do you mean `generic`?

**Review by Reto911 — APPROVED** [review]

**Pull request merged**